### PR TITLE
Add target docs for aarch64-apple-darwin (M1/M2 macs)

### PR DESCRIPTION
**Pull request by TimTheBig**
Add target documentation for `aarch64-apple-darwin`;  64-bit ARM-based macs running macOS, specifically Macs with "M" series Apple Silicon chips.
"M" series macs are quite popular so it would be useful to have some docs.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
r? Nilstrieb

File: src/doc/rustc/src/platform-support/aarch64-apple-darwin.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# `aarch64-apple-darwin`
+
+**Tier: 2 with host tools**
+
+64-bit ARM-based Apple devices running macOS, Macs with M1 or M2 Apple Silicon chips. M3 or higher is [`arm64e-apple-darwin`](platform-support/arm64e-apple-darwin.md).
```
Comment (pthariensflame) [inline src/doc/rustc/src/platform-support/aarch64-apple-darwin.md:5]: This isn't accurate; `aarch64-apple-darwin` works just fine on M3 too (I have one) and `arm64e-apple-darwin` works "just fine" (module ongoing ABI stability and implementation issues) on all M-series chips identically.  The distinction between the targets is the use of PAuth in the function call ABI, but PAuth is implemented on all of these CPUs and Apple's `arm64e` ABI is available on all of Apple's Arm-based devices, not just the Macs.

```suggestion
64-bit ARM-based Apple devices running macOS: Macs with M1-family or later Apple Silicon CPUs.
```

Comment (TimTheBig) [reply src/doc/rustc/src/platform-support/aarch64-apple-darwin.md]: That was a misunderstanding on my part. Do you happen to know who the maintainers are?

Comment (pthariensflame) [reply src/doc/rustc/src/platform-support/aarch64-apple-darwin.md]: Sadly, I don't; I would love to know myself.  :(

File: src/doc/rustc/src/platform-support/aarch64-apple-darwin.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+### Format
+
+The default binary format is Mach-O, the executable format used on Apple's platforms.
+
+## Building the target
+
+Just add it to the `target` with:
+```sh
+rustup target add aarch64-apple-darwin
```
Comment (pthariensflame) [inline src/doc/rustc/src/platform-support/aarch64-apple-darwin.md:25]: I think this section is supposed to be for building `rustc` such that it includes the target, not just for building code that uses the target.  Although looking at the other documents it's kind of inconsistently used?

**Comment by pthariensflame** [comment]
#124146 may be useful to note?

File: src/doc/rustc/src/platform-support/aarch64-apple-darwin.md (hunk 2 above)
Comment (TimTheBig) [reply src/doc/rustc/src/platform-support/aarch64-apple-darwin.md:25]: The command for building the target std is always the same, so I think this should be fine.

**Comment by madsmtm** [comment]
I've documented this target (and more) in https://github.com/rust-lang/rust/pull/124772 too.

**Comment by TimTheBig** [comment]
> I've documented this target (and more) in #124772 too.

Nice 👍, do you want to add this aarch64 doc to that pull.

**Comment by madsmtm** [comment]
> do you want to add this aarch64 doc to that pull.

Hmm, not sure what you mean by that? I've added documentation about it to the more general `platform-support/apple-darwin.md` file which also documents `x86_64-apple-darwin`.

**Comment by TimTheBig** [comment]
My mistake I misread. Your docs are good so I will close this pull, I look foreword to reading the new pages.

**Pull request closed without merging**